tests/basic/pure_test.go

```diff
@@ -5130,3 +5130,125 @@ var _ = Describe("{ValidatePodNameinVolume}", func() {
+			var pvcList []string
+			if isMultiTenancy {
+				_, err = pureutils.CreatePodinFA(faclient, podNameinFA)
+				log.FailOnError(err, fmt.Sprintf("Failed to create pod [%v] ", podNameinFA))
+				isPodExists, err := pureutils.IsPodExistsOnMgmtEndpoint(faclient, podNameinFA)
+				log.FailOnError(err, fmt.Sprintf("Failed to check if pod [%v] exists ", podNameinFA))
+				if !isPodExists {
+					log.FailOnError(fmt.Errorf("Pod [%v] is not created in FA", podNameinFA), "is pod created in FA?")
+				}
+				log.InfoD("Pod [%v] created ", podNameinFA)
+			}
+			Provisioner := fmt.Sprintf("%v", portworx.PortworxCsi)
+			context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+				AppKeys:            Inst().AppList,
+				StorageProvisioner: Provisioner,
+				PureFAPodName:      podNameinSC,
+				Namespace:          taskName,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", taskName)
+			contexts = append(contexts, context...)
+			ValidateApplications(contexts)
+			log.InfoD("waiting for a minute for volume name to populate")
+			for _, ctx := range contexts {
+				pvcList, err = GetVolumeNamefromPVC(ctx.App.NameSpace, pvcList)
+				log.FailOnError(err, "Failed to get volume name from PVC")
+			}
+			faErr := CheckVolumesExistinFA(flashArrays, pvcList, false)
+			log.FailOnError(faErr, "Failed to check if volumes created  exist in FA")
+			DestroyApps(contexts, nil)
+			faErr = CheckVolumesExistinFA(flashArrays, pvcList, true)
+			log.FailOnError(faErr, "Failed to check if volumes created  exist in FA")
+
+		}
+		stepLog = "Deploy FADA application with storage class having pure_fa_pod_name - this pod is under a realm in one of the arrays in pure.json"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			podNameinSC = "Torpedo-Test" + time.Now().Format("01-02-15h04m05s")
+			// In order to create a pod inside a realm , the pod name should be prefixed with realm, eg: <realm name>::<pod name>
+			PodNameinFA = realmName + "::" + podNameinSC
+			podCreateandAppDeploy(faWithRealm, PodNameinFA, podNameinSC, "fada-app-with-pod-realm", true)
+		})
+		stepLog = "Deploy FADA application with storage class not having pure_fa_pod_name"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			// Here we dont want to create any pod and create pure_fa_pod_name in storage class ,as this is standard way of Deploying Apps
+			podCreateandAppDeploy(faWithRealm, "", "", "fada-app-without-pod-realm", false)
+		})
+		stepLog = "Deploy FADA application with storage class having pure_fa_pod_name - this pod is not under a realm in one of the arrays in pure.json"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			podNameinSC = "Torpedo-Test" + time.Now().Format("01-02-15h04m05s")
+			// pod name will not contain any realm , so we just create pod with same naming convention in FA
+			PodNameinFA = podNameinSC
+			podCreateandAppDeploy(faWithoutRealm, PodNameinFA, podNameinSC, "fada-app-with-pod", true)
+		})
+
+	})
+
+})
```

Comment:
can we also enhance this test with more combinations?
on faWithRealm:
1. create pod without realm -- negative case
2. create pod under realm
3. create without pure_fa_pod_name

on faWithoutRealm:
1. create pod (no realm)
2. create pod under an existing realm (random) -- negative case
2. create without pure_fa_pod_name

Reply:
On further analysis with @sayalasomayajula-px ,
on faWithRealm:

create pod without realm -- negative case, need to add
create pod under realm -- already covered 
create without pure_fa_pod_name -- will be covered as part of existing fa regression jobs
on faWithoutRealm:

create pod (no realm) -- @Pure-AdamuKaapan can you confirm if FADA PVC with just pod name (without realm) is already covered?
create pod under an existing realm (random) -- negative case
create without pure_fa_pod_name -- will be covered as part of existing fa regression jobs